core/chains/evm/client/rpc_client.go

```diff
@@ -144,6 +128,10 @@ func NewRPCClient(
 		chainType:              chainType,
 		clientErrors:           cfg.Errors(),
 	}
+
+	r.Adapter = multinode.NewAdapter[RPCClient, *evmtypes.Head](
```

Comment:
why do we need to pass RPCClient into adapter? Can't we just inject latestBlock and latestFinalizedBlock methods?